### PR TITLE
vscode-extensions: uniform extension naming convention

### PR DESCRIPTION
**Pull request by ratsclub**
###### Motivation for this change

According to https://github.com/NixOS/nixpkgs/pull/127174#discussion_r653221472, we are trying to uniformize the case on `vscode-extensions`. However there are extensions that do not have all small-case characters. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ratsclub** [comment]
Weird that I can use the extension just fine when using my fork as the `input` of my `flake.nix`... Will fix it.

**Comment by ratsclub** [comment]
Wait, is the package supposed to break as it is a breaking change? :thinking:

**Comment by happysalada** [comment]
if you click on the CI failures details button, it points to line 1513 of default.nix from vscode-extensions
this is what it looks like
```
  aliases = self: super: {
    # aliases
    ms-vscode = lib.recursiveUpdate super.ms-vscode { inherit (super.golang) Go; };
  };
```

if you rename `Go` to `go`, it should fix that failure

Thank you for the PR!

**Comment by happysalada** [comment]
@SuperSandro2000 this looks good to me.
Maybe you will spot something that I missed?
(no worries if you are busy)

**Comment by happysalada** [comment]
Thank you again for the PR!

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 1]
```diff
@@ -142,6 +142,11 @@ pt-services.clipcat.enable).
 
 - the `mingw-64` package has been upgraded from 6.0.0 to 9.0.0
 
+- The following Visual Studio Code extensions were renamed to keep the naming convention uniform.
+  - `bbenoist.Nix` -> `bbenoist.nix`
+  - `CoenraadS.bracket-pair-colorizer` -> `coenraads.bracket-pair-colorizer`
+  - `golang.Go` -> `golang.go`
```
Comment (SuperSandro2000) [inline nixos/doc/manual/release-notes/rl-2111.section.md:148]: This is a useless release note. Just add aliases for them.

Comment (ratsclub) [reply nixos/doc/manual/release-notes/rl-2111.section.md:148]: Should I open another PR to fix this as this has been merged already?

Comment (SuperSandro2000) [reply nixos/doc/manual/release-notes/rl-2111.section.md:148]: You can leave the release notes like they are. I think adding aliases is worth it anyway. Best would it be if they are gated behind allowAliases.

**Comment by priegger** [comment]
"bbenoist.Nix" is mentioned in the comment in pkgs/applications/editors/vscode/with-extensions.nix, which should also be fixed, I think.

**Comment by ratsclub** [comment]
> "bbenoist.Nix" is mentioned in the comment in pkgs/applications/editors/vscode/with-extensions.nix, which should also be fixed, I think.

I'll try to add an alias and fix the other issues mentioned in this PR.

**Pull request merged**